File: scratch/check_schema.ts
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
import path from 'path';

dotenv.config({ path: path.resolve(process.cwd(), '.env.local') });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY!;
const supabase = createClient(supabaseUrl, supabaseKey);

async function checkSchema() {
  const { data, error } = await supabase.from('medicines').select('*').limit(1);
  if (error) {
    console.error('Error fetching medicines:', error);
  } else {
    console.log('Medicine columns:', Object.keys(data[0] || {}));
  }

  const { data: patientData, error: patientError } = await supabase.from('patients').select('*').limit(1);
  if (patientError) {
    console.error('Error fetching patients:', patientError);
  } else {
    console.log('Patient columns:', Object.keys(patientData[0] || {}));
  }
}

checkSchema();

File: scratch/check_schema.py
import os
from supabase import create_client
from dotenv import load_dotenv

load_dotenv('.env.local')
SUPABASE_URL = os.getenv("NEXT_PUBLIC_SUPABASE_URL")
SUPABASE_SERVICE_KEY = os.getenv("SUPABASE_SERVICE_ROLE_KEY")
supabase = create_client(SUPABASE_URL, SUPABASE_SERVICE_KEY)

# Query to check foreign keys
sql = """
SELECT
    tc.table_name, 
    kcu.column_name, 
    ccu.table_name AS foreign_table_name,
    ccu.column_name AS foreign_column_name,
    rc.delete_rule
FROM 
    information_schema.table_constraints AS tc 
    JOIN information_schema.key_column_usage AS kcu
      ON tc.constraint_name = kcu.constraint_name
      AND tc.table_schema = kcu.table_schema
    JOIN information_schema.constraint_column_usage AS ccu
      ON ccu.constraint_name = tc.constraint_name
      AND ccu.table_schema = tc.table_schema
    JOIN information_schema.referential_constraints AS rc
      ON rc.constraint_name = tc.constraint_name
WHERE tc.constraint_type = 'FOREIGN KEY' AND tc.table_name = 'prescription_details';
"""

# Since I can't run raw SQL via the client easily without an RPC, 
# I'll check if there's an RPC or I'll just update the script to be safe.
# Actually, I'll just update the cleanup script to delete details first.

print("Updating cleanup script to delete details first for safety.")
